api/src/term/_raw_value.rs

```diff
@@ -0,0 +1,158 @@
+//! Implementation of the so-called [`RawValue`](./struct.RawValue.html).
+
+use mownstr::MownStr;
+use std::hash::{Hash, Hasher};
+
+/// A raw value is a string possibly split into two parts.
+///
+/// The second part of the raw value is intended for some implementations of
+/// IRIs, storing both a "namespace" and a "suffix". For other kinds of term,
+/// the second part is by default `None`.
+#[derive(Debug, Copy, Clone, Eq)]
+pub struct RawValue<'a>(pub &'a str, pub Option<&'a str>);
+
+impl<'a> RawValue<'a> {
+    /// A new raw value from namespace and suffix.
+    ///
+    /// To create a raw value without suffix use `From<&str>` implementation.
+    pub fn new(ns: &'a str, suffix: &'a str) -> Self {
+        RawValue(ns, Some(suffix))
+    }
+    /// If this represents an IRI, check if its an absolute one.
+    ///
+    /// _Note:_ This function uses a heuristic that if the string contains a
+    /// colon `:` this might be a schema and therefore an absolute IRI. In
+    /// fact, there is no check that `self` represents a valid IRI.
```

Comment:
The term "heuristic" implies that it could sometimes be wrong, which is not the case (under the pre-condition that the two strings *do* concatenate to a valid IRI reference). So I propose to rephrase it like that:
```suggestion
    /// _Note:_ This function merely checks that the resulting string starts
    /// with an IRI scheme. It does *not* further check the validity of the
    /// full IRI, which is supposed to be done by the TTerm implementation.
```